lib/utils.rb

```diff
@@ -31,5 +32,31 @@ def Utils.paramhash_to_str params
   params.collect {|k, v| if (v == true) then ";#{k}" else ";#{k}=#{v}" end}.join("")
 end
 
+def Utils.check_route_matches_proxy route, proxy
+  regexp = /<sip:(.*?)@(.*?):(.*?);(.*?)>/i
+  proxy_address = proxy.instance_variable_get(:@ip)
```

Comment:
You can just use `proxy.ip` and `proxy.port` - no need for instance_variable_get as they're exposed with `attr_reader`.